src/Meditation.UI/Controllers/PatchProcessController.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Meditation.AttachProcessService.Models;
+using Meditation.AttachProcessService;
+using Meditation.PatchingService;
+using Meditation.UI.Configuration;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading;
+using System;
+using System.IO;
+using CommunityToolkit.Mvvm.Input;
+using Meditation.UI.Controllers.Utils;
+
+namespace Meditation.UI.Controllers
+{
+    public partial class PatchProcessController
+    {
+        private readonly ApplicationConfiguration _configuration;
+        private readonly IAttachedProcessContext _attachedProcessContext;
+        private readonly IProcessListProvider _processListProvider;
+        private readonly IPatchApplier _patchApplier;
+
+        public PatchProcessController(
+            ApplicationConfiguration configuration,
+            IProcessListProvider processListProvider,
+            IAttachedProcessContext attachedProcessContext,
+            IPatchApplier patchApplier)
+        {
+            _configuration = configuration;
+            _processListProvider = processListProvider;
+            _attachedProcessContext = attachedProcessContext;
+            _patchApplier = patchApplier;
+        }
+
+        [RelayCommand]
+        public async Task ApplyPatch(string patchAssemblyPath, CancellationToken ct)
+        {
+            try
+            {
+                await ApplyPatchAsync(patchAssemblyPath, ct);
+                await DialogUtilities.ShowMessageBox(
+                    title: "Success",
+                    content: $"Applied patch \"{patchAssemblyPath}\" into target process.");
+            }
+            catch (Exception exception)
+            {
+                await DialogUtilities.ShowUnhandledExceptionMessageBox(exception);
+            }
+        }
+
+        [RelayCommand]
+        public Task ReversePatch(string patchAssemblyPath, CancellationToken ct)
+        {
+            try
+            {
+                // TODO: implement patch reverse

```

Comment:
[ToFix] Issue number

Reply:
This is a temporary TODO. It is already implemented here: #54